syntax = "proto3";

package detection;

message BBox {
  int32 xmin = 1;
  int32 ymin = 2;
  int32 width = 3;
  int32 height = 4;
}

message ObjectInfo {
  int32 track_id = 1;
  int32 class_id = 2;
  float confidence = 3;
  BBox bbox = 4;
  float distance = 5;
  int32 center_x = 6;
  int32 center_y = 7;
}

message ObjectList {
  repeated ObjectInfo objects = 1;
}

service DetectionService {
  rpc SendDetections (ObjectList) returns (DetectionResponse);
}

message DetectionResponse {
  bool success = 1;
  string message = 2;
}
